File: docker_config.go
package main

import (
	"path/filepath"
)

// DockerConfig represents Docker configuration for R Markdown rendering
type DockerConfig struct {
	Image   string
	Command []string
	Binds   []string
}

// CreateDockerConfig creates Docker configuration for rendering R Markdown
func CreateDockerConfig(rmdDir, inputFile string, outputFormat string) DockerConfig {
	// Inside the container, the file will be at /rmd/filename
	containerInputFile := filepath.Join("/rmd", filepath.Base(inputFile))

	// Determine the render command based on output format
	renderCommand := "render"
	if outputFormat == "pdf" {
		renderCommand = "render_pdf"
	} else if outputFormat == "word" {
		renderCommand = "render_word"
	}

	return DockerConfig{
		Image:   "r-server-rmd",
		Command: []string{renderCommand, containerInputFile},
		Binds:   []string{rmdDir + ":/rmd"},
	}
}

// CreateDockerComposeEnv generates docker-compose environment variables
func CreateDockerComposeEnv(rmdDir, inputFile string, outputFormat string) map[string]string {
	// Inside the container, the file will be at /rmd/filename
	containerInputFile := filepath.Join("/rmd", filepath.Base(inputFile))

	return map[string]string{
		"RMD_DIR":       rmdDir,
		"INPUT_FILE":    containerInputFile,
		"OUTPUT_FORMAT": outputFormat,
	}
}

File: src/docker-config.ts
import * as path from "path";

/**
 * Docker configuration for R Markdown rendering
 */
export interface DockerConfig {
  image: string;
  command: string[];
  binds: string[];
}

/**
 * Create Docker configuration for rendering R Markdown
 * 
 * @param rmdDir Directory containing R Markdown files
 * @param inputFile Path to the R Markdown file to render
 * @param outputFormat Output format (html, pdf, or word)
 * @returns Docker configuration
 */
export function createDockerConfig(
  rmdDir: string,
  inputFile: string,
  outputFormat: "html" | "pdf" | "word" = "html"
): DockerConfig {
  // Inside the container, the file will be at /rmd/filename
  const containerInputFile = path.join("/rmd", path.basename(inputFile));
  
  // Determine the render command based on output format
  const renderCommand = outputFormat === "html" 
    ? "render" 
    : outputFormat === "pdf" 
      ? "render_pdf" 
      : "render_word";
  
  return {
    image: "r-server-rmd",
    command: [renderCommand, containerInputFile],
    binds: [`${rmdDir}:/rmd`]
  };
}

/**
 * Generate docker-compose environment variables
 * 
 * @param rmdDir Directory containing R Markdown files
 * @param inputFile Path to the R Markdown file to render
 * @param outputFormat Output format (html, pdf, or word)
 * @returns Environment variables for docker-compose
 */
export function createDockerComposeEnv(
  rmdDir: string,
  inputFile: string,
  outputFormat: "html" | "pdf" | "word" = "html"
): Record<string, string> {
  // Inside the container, the file will be at /rmd/filename
  const containerInputFile = path.join("/rmd", path.basename(inputFile));
  
  return {
    RMD_DIR: rmdDir,
    INPUT_FILE: containerInputFile,
    OUTPUT_FORMAT: outputFormat
  };
}
